analyses/project-updates/01-generate-project-report.Rmd

```diff
@@ -0,0 +1,420 @@
+cellranger_results_dir <- file.path(cellranger_module_results_dir, "03_cellranger_count_summary", cellranger_parameters, glue::glue("multi_run_{cellranger_parameters}"))
+#seurat_plots_dir <- file.path(module_plots_dir, "02_Seurat_qc") 
+SoupX_results_dir <- file.path(module_results_dir, "01_SoupX") 
+#SoupX_plots_dir <- file.path(module_plots_dir, "01_SoupX") 
+scDblFinder_results_dir <- file.path(module_results_dir, "03_scDblFinder") 
+#scDblFinder_plots_dir <- file.path(module_plots_dir, "03_scDblFinder") 
+Filter_object_results_dir <- file.path(module_results_dir, "04_Filter_object") 
+#Filter_object_plots_dir <- file.path(module_plots_dir, "04_Filter_object") 
+
+# Input files
+multiqc_general_stats_results_file <- file.path(fastqc_results_dir, "multiqc_general_stats.txt")
+multiqc_fastqc_results_file <- file.path(fastqc_results_dir, "multiqc_fastqc.txt")
+fastqc_overrepresented_results_file <- file.path(fastqc_results_dir, "fastqc_overrepresented_sequences_plot.txt")
+
+cellranger_results_file <- file.path(cellranger_results_dir, "QC_Summary_CellRanger_Report.tsv")
+SoupX_results_file <- file.path(SoupX_results_dir, "SoupX_contamination_summary.tsv")
+scDblFinder_results_file <- file.path(scDblFinder_results_dir, "doublets_metrics.tsv")
+cells_table_file <- file.path(Filter_object_results_dir, "number_of_cells_per_sample.tsv")
+#metadata_file <- file.path(metadata_dir, "project_metadata.tsv") # metadata input file
+
+summary_samples_table_file <- file.path(metadata_dir, metadata_file) # summary_samples_table file
+```
+
+
+# Methods
+
+## Data
+
+### `r PROJECT_NAME`
+
+```{r read-table, echo = FALSE}
+summary_samples_table <- readr::read_tsv(summary_samples_table_file, guess_max = 100000, show_col_types = FALSE)
+```
+
+```{r, fig.align = "left", results = "asis", message = FALSE, warning = FALSE, echo = FALSE}
+# Create the data frame
+tables1 <- summary_samples_table
+cat("  \n<div align=\"center\" style=\"font-size:80%\">  \n")
+print(knitr::kable(tables1, align = "lcccc", caption = "Summary of samples in the cohort"))
+cat("  \n</div>  \n")
+cat("  \n\\pagebreak  \n")
+```
+
+  
+# Results
+
+All results can be found here:
+
+- Cluster users can access data from: `r root_dir`/analyses/
+- Mac users can access data from: `r mac_results_filepath`/`r cohort_value`/analyses/
+- `r results_filepath`/`r cohort_value`/analyses/
+
+
+We used FastQC (v0.11.9) and multiqc (v1.25) to perform quality control tool for all libraries across the project from the raw sequencing reads. Next, we aligned these reads using CellRanger (v8.0.1) (10x Flex Genomics). 
+
+## fastqc-analysis
+
+FastQC metrics such as per-base sequence quality, GC content, and adapter contamination were within expected thresholds, ensuring that the data is suitable for downstream analyses. Further validation steps, such as alignment or feature counting, will help confirm that the data is suitable for further processing and analysis.
```

Comment:
Is this another section that should be edited by the user for their specific data? It states that metrics "were within expected thresholds", but that may not always be true. If this is supposed to be edited by the user, maybe ensure that is documented clearly somewhere for this module. And if it is not supposed to be edited by the user, then perhaps this section could be changed to specify that these are some of the metrics we are checking during this step to determine the quality of the sequencing.

Reply:
```suggestion
FastQC metrics such as per-base sequence quality, GC content, and adapter contamination can indicate the quality of the sequencing data and contribute to determining if the data is suitable for downstream analyses. Further validation steps, such as alignment or feature counting, will also help confirm if the data is suitable for further processing and analysis.
```